packages/components/src/unit-control/index.tsx

```diff
@@ -92,24 +106,24 @@ function UnitControl(
 		setUnit( next );
 	};
 
-	const mayUpdateUnit = ( event ) => {
-		if ( ! isNaN( event.target.value ) ) {
+	const mayUpdateUnit = ( event: SyntheticEvent< HTMLInputElement > ) => {
+		if ( ! isNaN( Number( event.currentTarget.value ) ) ) {
 			refParsedValue.current = null;
 			return;
 		}
 		const [ parsedValue, parsedUnit ] = getValidParsedUnit(
-			event.target.value,
+			event.currentTarget.value,
 			units,
 			value,
 			unit
 		);
 
-		refParsedValue.current = parsedValue;
+		refParsedValue.current = parsedValue?.toString();
```

Comment:
Our `Value` type accepts a number as a valid value. However, the Input element itself should only be passed a string, so here we're treating the value as a string so that the state reducer and input element are happy.

The `handleOnChange` method still calls the passed in `onChange` with the real _parsed_ value, though, so this shouldn't affect consumers of the component 🤞

Reply:
Makes sense, thank you for the explanation!z